client/src/features/bulletins/Bulletins/Filters/DoubleSlider/DoubleSlider.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { useRef, useEffect } from 'react';
+import './DoubleSlider.scss';
+import { CustomInput } from 'reactstrap';
+
+type Props = {
+  setMinPrice: (minPrice: number) => void;
+  setMaxPrice: (maxPrice: number) => void;
+};
+
+export const DoubleSlider = React.memo(({ setMinPrice, setMaxPrice }: Props) => {
+  const minSlider = useRef<HTMLInputElement>(null);
+  const minBubble = useRef<HTMLInputElement>(null);
+
+  const maxSlider = useRef<HTMLInputElement>(null);
+  const maxBubble = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    updateBubbles(minSlider.current!, minBubble.current!);
+    updateBubbles(maxSlider.current!, maxBubble.current!);
+  }, []);
+
+  const updateBubbles = (cursorRef: HTMLInputElement, bubbleRef: HTMLInputElement) => {
+    if (cursorRef && bubbleRef) {
+      const newValue =
+        (Number(cursorRef.value) * 100) / (Number(cursorRef.max) - Number(cursorRef.min));
+      const newPosition = 103 - newValue * 1.2;
+      bubbleRef.innerHTML = `<span>${cursorRef.value}</span>`;
+      bubbleRef.style.left = `calc(${newValue}% + (${newPosition}px))`;
+    }
+  };
+
+  const avoidOverlay = (e: React.ChangeEvent) => {
+    if (maxSlider.current && minSlider.current) {
+      const _max = maxSlider.current;
+      const _min = minSlider.current;
+      if (_min.valueAsNumber + 1 >= _max.valueAsNumber && e.currentTarget.id === _max.id) {
+        _min.valueAsNumber = _max.valueAsNumber - 1;
```

Comment:
Madre de Dios.
![](https://images.vexels.com/media/users/3/143355/isolated/preview/7c9190c0fa15ec714bd3bea70c0ed2bb-madre-de-dios-cara-de-meme-by-vexels.png)

Mai settare nuovi campi su oggetti esistenti, per di più su oggetti nativi del DOM. Mi chiedevo infatti perché non avessi mai sentito parlare di `valueAsNumber` come proprietà nativa del DOM. Se proprio devi farlo, che nel tuo caso è già mai perché non hai ancora l'esperienza per valutare quando veramente serve, almeno mettici un underscore davanti per far capire che è una proprietà custom. Comunque in questo caso devi convertire back-and-forth tra stringa e numero. In realtà quando passi come `value={min}` come numero viene convertito implicitamente a stringa senza problemi, quindi ti basta gestire da string a number al change